### PR TITLE
Fix message on sign typed data v4 screen

### LINKED ISSUE #16400: signTypedData_v4 message doesn't render
[Report on  @slack](https://consensys.slack.com/archives/CN2H44RAM/p1667596319572549)

Some messages don't render on signTypedData_v4 confirmation screen (see screenshot). 

Screen Shot 2022-11-04 at 2 16 23 PM (1)

It might be a bug related with fixed-size and nested dynamic arrays. Here are two examples of messages that don't render:
```
type => {"A": [{"name": "b", "type": "uint256[2]"}]}
data => {"b": [123, 456]}
```
```
type => {"A": [{"name": "b", "type": "uint256[][]"}]}
data => {"b": [[12, 34], [56, 78]]}
```
Additional info: `{"name":"tree","type":"OrderComponents[2]"}`  does not render, but a single OrderComponent (which itself includes dynamic arrays of structs) renders fine.

### PR DESCRIPTION
**Pull request by amerkadicE**
## Explanation

We have an issue when sanitize the message on the sign-typed data v4 confirmation screen. Current logic did not support sanitizing nested arrays and arrays with a length defined.

This PR it will support both nested arrays and arrays with length so the message will render correctly.

 Fixes #16400 

### Before

image

### After
image

## Manual Testing Steps

1. Clone the test dapp
2. Change the following lines
 line 1160 primaryType: 'Test',
3. underneath line 1161, add a new type with the params given in this ticket above
 Test: [
           {"name": "b", "type": "uint256[2]"}
         ],
4. underneath 1156, add the test data for this type: b: [123, 456],
5. Start the project npm run start
6. Connect to MetaMask
7. Click Sign on Type 4
8. Note that message will render

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/helpers/utils/util.js [hunk 1]
```diff
@@ -366,6 +366,7 @@ const solidityTypes = () => {
     'bytes',
     'int',
     'uint',
+    'uint256',
```
Comment (FrederikBolding) [inline ui/helpers/utils/util.js:369]: This should already be included in `uints` below?

Comment (amerkadicE) [reply ui/helpers/utils/util.js]: You are correct, I updated my PR.

**Comment by mirjanaKukic** [comment]
Verified by QA

**Review by brad-decker — APPROVED** [review]

File: ui/helpers/utils/util.js [hunk 2]
```diff
@@ -411,11 +411,37 @@ const solidityTypes = () => {
...
   if (!types) {
     throw new Error(`Invalid types definition`);
   }
 
+  // Primary type can be an array.
+  const isArray = primaryType && isArrayType(primaryType);
+  if (isArray) {
+    return msg.map((value) =>
+      sanitizeMessage(value, stripOneLayerofNesting(primaryType), types),
+    );
```
Comment (digiwand) [inline ui/helpers/utils/util.js:437]: 💯

**Review by digiwand — APPROVED** [review]

File: ui/helpers/utils/util.js (hunk 2 above)
Comment (FrederikBolding) [reply ui/helpers/utils/util.js:437]: 😎

**Pull request merged**